Update install notes (#749)
CrayLabs/SmartSim: doc/installation_instructions/basic.rst

Basic Installation
******************

> The following instructions serve as a guide for installing both SmartSim and

Reviewer: ```suggestion
The following instructions guide you through installing SmartSim and
```

.. note::

  GCC is recommended to build the backends for SmartSim. CUDA 11.8 requires GCC
  9 or 11, CUDA 12 requires GCC 11 or higher. SmartRedis can be compiled with

>   GCC, Intel, Cray, and Nvidia compilers.

Reviewer: Consider reformatting for clarification of the GCC-specific requirements

```
GCC is recommended to build the backends for SmartSim.
- CUDA 11.8 requires GCC 9 or 11.
- CUDA 12 requires GCC 11 or higher.

SmartRedis can be compiled with GCC, Intel, Cray, and Nvidia compilers.
```

ML Library Support
==================

> We currently support both Nvidia and AMD GPUs when using RedisAI for GPU

Reviewer: ```suggestion
SmartSim supports using Nvidia and AMD GPUs with RedisAI for GPU
```

> inference. The support for these GPUs often depends on the version of the CUDA

Reviewer: ```suggestion
inference. GPU support often depends on the version of the CUDA
```

> or ROCm stack that is availble on your machine. In _most_ cases, the versions of

Reviewer 2: availble -> available

the ML frameworks are backwards compatible. If you encounter problems, please

> contact us and we can build the backend libraries for your desired version of

Reviewer 2: Add a link point pointing to SS contact information

> CUDA and ROCm.

Reviewer 2: CUDA and ROCm -> CUDA or ROCm

CPU backends are provided for Apple (both Intel and Apple Silicon) and Linux (x86_64).

> Be sure to reference the table below to find which versions of the ML libraries

Reviewer: ```suggestion
Please reference the table below to find which versions of the ML libraries
```

are supported for your particular platform. Additionally, see :ref:`Platform

> Installation Guide <platform-installation>` for helpful information regarding

Reviewer 2: `regarding for specific systems` - `regarding specific systems`

.. note::

  We suggest using GCC to build Redis, RedisAI, and the ML backends. For specific

>   version requirements see the :ref:`Requirements <requirements>` section.

Reviewer 2: it does not look like there is a Requirements section, and it instead takes you to the ML Library Support/Linux section -> should this link instead point to the ML Library Support section?

  SmartRedis can be compiled with GCC, Intel, Cray, and Nvidia compilers.

ML Library Support
==================

> SmartSim supports using Nvidia and AMD GPUs when using RedisAI for GPU

Reviewer 2: `SmartSim supports using Nvidia and AMD GPUs when using RedisAI for GPU inference` - what are your thoughts on rephrasing it to `SmartSim enables the use of Nvidia and AMD GPUs for GPU inference with RedisAI`?

inference. GPU support often depends on the version of the CUDA or ROCm stack

> that is available on your machine. In _most_ cases, the versions of the ML

Reviewer 2: It looks like _most_ is not rendering correctly in readthedocs and is showing up exactly as is - instead try `*most*`

frameworks are backwards compatible. If you encounter problems, please contact

> us at (smartsim at hpe dot com) and we can build the backend libraries for your

Reviewer 2: for pointing them on how to contact us, what are your thoughts on pointing them to this area of the docs?

https://www.craylabs.org/docs/contributing.html#how-to-connect

so they have multiple options?

ALSO! what are your thoughts on changing (smartsim at hpe dot com) to instead 
`smartsim@hpe.com <mailto:smartsim@hpe.com>`_

> 

Reviewer 2: I notice in the Linux tabs, there are additional requirements for CUDA 11 and CUDA 12 but not for ROCm or CPU - just raising this just incase!

Reviewer 2: oh also why does ROCm 6 have N/A for two columns?

Reviewer 2: CPU seems a little out of place, what are your thoughts on splitting this table into two with
1. GPU Configurations
2. CPU Configurations
